### PR TITLE
SONAR-9745 Drop support of Microsoft SQLServer 2012

### PR DESCRIPTION
**Pull request by simonbrandhof**

### CODE REVIEW

File: server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java [hunk 1]
```diff
@@ -108,6 +109,23 @@ public void test_mysql() throws Exception {
   }
 
   @Test
+  public void mssql_2012_is_not_supported() throws Exception {
+    expectedException.expect(MessageException.class);
+    expectedException.expectMessage("Unsupported mssql version: 11.0. Minimal supported version is 12.0.");
+
+    Database db = mockDb(new MsSql(), 11, 0, "6.1");
```
Comment (sns-seb) [inline server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java]: "6.1" is a bad copy/paste, right?

File: server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java [hunk 2]
```diff
@@ -108,6 +109,23 @@ public void test_mysql() throws Exception {
...
+    new DatabaseChecker(db).start();
+    // no error
+  }
+
+  @Test
+  public void mssql_2014_is_supported() throws Exception {
+    Database db = mockDb(new MsSql(), 12, 0, "6.1");
```
Comment (sns-seb) [inline server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java]: "6.1" is a bad copy/paste, right?

Comment (sns-seb) [inline server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java]: a test that an intermediary version such as "12.0.4100.1" would be nice

File: server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java (hunk 1 above)
Comment (simonbrandhof) [inline server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java]: nope, it's version of MS driver

File: server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java (hunk 2 above)
Comment (simonbrandhof) [inline server/sonar-db-core/src/test/java/org/sonar/db/DatabaseCheckerTest.java]: in fact only the major and minor versions are loaded from JDBC metadata. Complete version is not used.

**Pull request closed without merging**